context/app/static/js/pages/SavedList/SavedList.jsx

```diff
@@ -20,22 +21,36 @@ function getListAndItsEntities(savedLists, listTitle) {
 
 function SavedList({ listTitle }) {
   const decodedTitle = decodeURIComponent(listTitle);
+  const [editedListTitle, setEditedListTitle] = useState(false);
+
+  const currentTitle = editedListTitle || decodedTitle;
 
   const { savedLists } = useSavedEntitiesStore(usedSavedEntitiesSelector);
-  const [savedList, listEntities] = getListAndItsEntities(savedLists, decodedTitle);
+  const [savedList, listEntities] = getListAndItsEntities(savedLists, currentTitle);
+
+  const { description } = savedList;
   return (
     <>
       <Typography variant="subtitle1" component="h1" color="primary">
         List
       </Typography>
-      <Typography variant="h2">{decodedTitle}</Typography>
+      <Typography variant="h2">{currentTitle}</Typography>
       <RightAlignedButtonRow
         leftText={
           <Typography variant="body1" color="primary">
             {Object.keys(listEntities).length}
           </Typography>
         }
-        buttons={<SavedListMenuButton listTitle={decodedTitle} />}
+        buttons={
```

Comment:
This might also be more idiomatic as children?

Reply:
As we discussed, in situations where there are multiple groups of components that need to be passed down, components as props can make sense but prefer children where possible.